ENIGMAsystem/SHELL/Platforms/SDL/Event.cpp

```diff
@@ -210,16 +211,27 @@ void SDL_Event_Handler::windowResized(const SDL_Event *event) {
   enigma_user::draw_clear(enigma_user::window_get_color());
 }
 
-void SDL_Event_Handler::joyDeviceAdded(const SDL_Event *event) { addGamepad(event->cdevice.which); }
+// map of joystick instance ids to device indexes
+std::unordered_map<SDL_JoystickID,int> joystickDevices;
```

Comment:
No need to store int internally; store a pointer to the dense joystick storage to remove an extra array lookup.

Reply:
That doesn't work because our functions look up the state cached on the array. What I mean is, if I stored the pointer you are talking about, I wouldn't be able to mutate the thing that you are talking about. Also that doesn't work for removing, because then you don't know which user array to remove.
https://github.com/enigma-dev/enigma-dev/blob/ad62bafd62c8badff106c32e01b6022b7f23ba82/ENIGMAsystem/SHELL/Platforms/SDL/Gamepad.cpp#L168